types/ejs/index.d.ts

```diff
@@ -54,82 +59,401 @@ export type RenderFileCallback<T> = (err: Error, str?: string) => T;
-    localsName?: string;
-    /**
-     * Remove all safe-to-remove whitespace, including leading and trailing whitespace.
-     * It also enables a safer version of `-%>` line slurping for all scriptlet tags (it does not strip new lines of tags in the middle of a line).
-     */
-    rmWhitespace?: boolean;
-    /**
-     * The escaping function used with `<%=` construct.
-     * It is used in rendering and is `.toString()`ed in the generation of client functions.
-     * (By default escapes XML).
-     */
-    escape?(str: string): string;
-    /** Whether or not to use a Promise */
-    async?: boolean;
-}
+/**
+ * EJS template function cache. This can be a LRU object from lru-cache
+ * NPM module. By default, it is `utils.cache`, a simple in-process
+ * cache that grows continuously.
+ */
+export let cache: Cache;
+
+/**
+ * Custom file loader. Useful for template preprocessing or restricting access
+ * to a certain part of the filesystem.
+ *
+ * @param path the path of the file to be read
+ * @return the contents of the file as a string or object that implements the toString() method
+ *
+ * @default fs.readFileSync
+ */
+export type fileLoader = (path: string) => string | { toString(): string };
+export let fileLoader: fileLoader;
+
+/**
+ * Name of the object containing the locals.
+ *
+ * This variable is overridden by {@link Options}`.localsName` if it is not
+ * `undefined`.
+ *
+ * @default 'locals'
+ */
+export let localsName: string;
+
+/**
+ * The opening delimiter for all statements. This allows you to clearly delinate
+ * the difference between template code and existing delimiters. (It is recommended
+ * to synchronize this with the closeDelimiter property.)
+ *
+ * @default '<'
+ */
+export let openDelimiter: string;
+
+/**
+ * The closing delimiter for all statements. This allows to to clearly delinate
+ * the difference between template code and existing delimiters. (It is recommended
+ * to synchronize this with the openDelimiter property.)
+ *
+ * @default '>'
+ */
+export let closeDelimiter: string;
+
+/**
+ * The delimiter used in template compilation.
+ *
+ * @default '%'
+ */
+export let delimiter: string;
+
+/**
+ * Promise implementation -- defaults to the native implementation if available
+ * This is mostly just for testability
+ *
+ * @default Promise
+ */
+export let promiseImpl: PromiseConstructorLike | undefined;
 
-export function escapeRegexChars(s: string): string;
 /**
  * Escape characters reserved in XML.
  *
- * This is simply an export of {@link module:utils.escapeXML}.
+ * This is simply an export of `utils.escapeXML`.
  *
  * If `markup` is `undefined` or `null`, the empty string is returned.
  */
-export function escapeXML(markup: string): string;
-export interface Cache {
-    set(key: string, val: TemplateFunction): void;
-    get(key: string): TemplateFunction | undefined;
+export function escapeXML(markup?: any): string;
+
+export class Template {
+	/**
```

Comment:
I think four-space tabs are preferred in this repo.  I'm a little surprised there isn't a lint error.

Reply:
Tabs&nbsp;are&nbsp;allowed, and&nbsp;they’re the&nbsp;superior indentation&nbsp;codepoint, as&nbsp;in&nbsp;**UTF‑8** it’s&nbsp;1&nbsp;byte per&nbsp;tab vs&nbsp;a&nbsp;ton&nbsp;for&nbsp;spaces.

With&nbsp;tabs, it’s&nbsp;also&nbsp;possible to&nbsp;customise the&nbsp;size, whereas&nbsp;with&nbsp;spaces, you’re&nbsp;forced to&nbsp;use&nbsp;whichever indentation&nbsp;size the&nbsp;project tries&nbsp;to&nbsp;cram down&nbsp;your&nbsp;throat.

Reply:
Fair enough, but I think we can agree that worst option is a mix of spaces and tabs.  In this repo, I believe we've decided to standardize on spaces (though it has not been universally rolled out, to avoid excessive churn).

Reply:
Last&nbsp;I&nbsp;checked, package&nbsp;maintainers could&nbsp;choose to&nbsp;use&nbsp;whichever they&nbsp;preferred.

Reply:
I'll merge it, but be aware that they may be converted to spaces, if the convention becomes an explicit policy.